src/software/ai/navigator/path_planner/enlsvg_path_planner_test.cpp

```diff
@@ -657,31 +657,31 @@ TEST_F(TestEnlsvgPathPlanner,
 
     ASSERT_TRUE(path != std::nullopt);
 
-    std::vector<Point> path_points = path->getKnots();
+    std::vector<Point> path_points = path.value();
 
     // Make sure the start and end points match
-    EXPECT_EQ(start, path->getStartPoint());
-    EXPECT_EQ(dest, path->getEndPoint());
+    EXPECT_EQ(start, path.value().front());
+    EXPECT_EQ(dest, path.value().back());
 
     // Make sure path does not exceed a bounding box
     Rectangle bounding_box({-2.1, -1.1}, {4.1, 4.1});
     TestUtil::checkPathDoesNotExceedBoundingBox(path_points, bounding_box);
-    TestUtil::checkPathDoesNotIntersectObstacle(path->getKnots(), obstacles);
+    TestUtil::checkPathDoesNotIntersectObstacle(path.value(), {obstacles});
```

Comment:
I suspect `{obstacles}` won't work here because it's already a vector of ObstaclePtrs. I could be wrong though. Could you check if this test is clean and completely passes?